app/services/analytics_service.rb

```diff
@@ -1,6 +1,7 @@
 class AnalyticsService
-  def initialize(user_or_org, start_date: "", end_date: "")
+  def initialize(user_or_org, start_date: "", end_date: "", single_article_id: nil)
```

Comment:
Why not just `article_id` ?

Reply:
I thought it was too similar to `article_ids` but let me know if you think it's more readable / less confusing than `single_article_id`.

Reply:
`article_ids` is internal business logic, a user of the class won't know about it, honestly I suggested `article_id` because is less typing than `single_article_id` since there no "double_article_id" 😂